x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/tree/Source.java

```diff
@@ -51,6 +51,21 @@ public static <S extends StreamInput & PlanStreamInput> Source readFrom(S in) th
         return new Source(new Location(line, charPositionInLine), text);
     }
 
+    /**
+     * Read the components of a {@link Source} and throw it away, returning
+     * {@link Source#EMPTY}. Use this when you will never use the {@link Source}
+     * and there is no chance of getting a {@link PlanStreamInput}.
+     */
+    public static Source readEmpty(StreamInput in) throws IOException {
+        if (in.readBoolean() == false) {
+            return EMPTY;
+        }
+        in.readInt();
+        in.readInt();
+        in.readInt();
```

Comment:
Thought: Maybe we should abstract reading the three ints into a `readOptionalSourcePosition` or so (whose return value we discard)?

That is duplicated in `readFrom`; factoring into a method would have a higher chance of staying in sync with `readFrom`